src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/DirectoryInfo/CreateSubdirectory.cs

```diff
@@ -235,6 +236,117 @@ public void ParentDirectoryNameAsPrefixShouldThrow()
             Assert.Throws<ArgumentException>(() => di.CreateSubdirectory(Path.Combine("..", randomName + "abc", GetTestFileName())));
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.IsSubstAvailable))]
+        public void CreateSubdirectoryFromRootDirectory_Windows()
+        {
+#if TARGET_WINDOWS
+            // On Windows, use VirtualDriveHelper to create a test root directory
+            using VirtualDriveHelper virtualDrive = new();
+            char driveLetter = virtualDrive.VirtualDriveLetter;
+            string rootPath = $"{driveLetter}:\\";
+            
+            DirectoryInfo rootDir = new DirectoryInfo(rootPath);
+            string subDirName = GetTestFileName();
+            
+            // This should work without throwing ArgumentException
+            DirectoryInfo result = rootDir.CreateSubdirectory(subDirName);
+            
+            Assert.NotNull(result);
+            Assert.Equal(Path.Combine(rootPath, subDirName), result.FullName);
+            Assert.True(result.Exists);
+            
+            // Clean up
+            result.Delete();
+#endif
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void CreateSubdirectoryFromRootDirectory_Unix()
+        {
+            // Skip if not on Unix or not privileged
+            if (PlatformDetection.IsWindows || !PlatformDetection.IsPrivilegedProcess)
+                return;
+                
+            // On Unix, create a temporary directory and test subdirectory creation
+            // This is a simplified test that avoids chroot but still tests the core functionality
+            string tempRoot = Path.Combine(Path.GetTempPath(), "test_root_" + Path.GetRandomFileName());
+            Directory.CreateDirectory(tempRoot);
+            
+            try
+            {
+                // Create a directory that acts as our test root
+                DirectoryInfo testRootDir = new DirectoryInfo(tempRoot);
+                string subDirName = GetTestFileName();
+                
+                // Test that CreateSubdirectory works on our test root
+                DirectoryInfo result = testRootDir.CreateSubdirectory(subDirName);
+                
+                Assert.NotNull(result);
+                Assert.Equal(Path.Combine(tempRoot, subDirName), result.FullName);
+                Assert.True(result.Exists);
+                
+                // Clean up
+                result.Delete();
+            }
+            finally
+            {
+                try
+                {
+                    Directory.Delete(tempRoot, recursive: true);
+                }
+                catch
+                {
+                    // Ignore cleanup errors
+                }
+            }
```

Comment:
This needs to be something like: 
```
[DllImport("libc", SetLastError = true)]
private static extern int chroot(string path);

...

RemoteExecutor.Invoke(() =>
{
    string newRoot = Directory.CreateTempSubdirectory("new_root").FullName;
    chroot(newRoot);

    // Test CreateSubdirectory... no need to cleanup since this is a temp folder.
}).Dispose();
```